pkg/storage/file/backend.go

```diff
@@ -708,6 +699,18 @@ func (backend *Backend) setOptionsLocked(
 	recordType string,
 	options *databrokerpb.Options,
 ) error {
+	existingOpts := backend.options[recordType]
+	if existingOpts != nil && len(existingOpts.GetIndexableFields()) > 0 {
```

Comment:
If we added an indexable field, and then removed it, shouldn't we also delete the associated index values? So I think we'd either need to always delete.. or maybe just check if the indexed fields changed?